dmtcp/src/common/shm_config.c

```diff
@@ -14,3 +14,26 @@ void mc_get_shm_handle_name(char *dst, size_t sz) {
   snprintf(dst, sz, "/mcmini-%s-%lu", getenv("USER"), (long)getppid());
   dst[sz - 1] = '\0';
 }
+
+// Define the global variable and initialize the mutex
+volatile void* global_shm_start = NULL;
+pthread_mutex_t shm_start_mutex = PTHREAD_MUTEX_INITIALIZER;
+
+// Function to set the global shared memory start address
+void set_global_shm_start(void* addr) {
+    pthread_mutex_lock(&shm_start_mutex);
+    global_shm_start = addr;
+    pthread_mutex_unlock(&shm_start_mutex);
+}
+
+// Function to get the global shared memory start address
+volatile void* get_global_shm_start() {
+    // Use a thread-local variable to cache the value
+    static __thread volatile void* local_shm_start = NULL;
+
+    if (local_shm_start == NULL) {
+        local_shm_start = global_shm_start;
+    }
```

Comment:
This needs to be wrapped in a `pthread_mutex_lock()/pthread_mutex_unlock()`, but otherwise it's good

Reply:
When I say `this`, I meant only line 35 to be precise since `local_shm_start` is thread_local.